protocol/app/prepare/other_msgs.go

```diff
@@ -46,6 +54,16 @@ func RemoveDisallowMsgs(ctx sdk.Context, decoder sdk.TxDecoder, txs [][]byte) []
 
 	var filteredTxs [][]byte
 	for i, txBytes := range txs {
+		// If tx is index 0 and VE enabled, this ts is a VE tx and should be decoded with the extCommitCodec.
+		if i == constants.OracleInfoIndex && ve.VoteExtensionsEnabled(ctx) {
```

Comment:
Should we be removing any other transactions that decode into `extCommitInfo`? for instance if a proposer attempts to include an extended commit tx later in the block as well?

Reply:
They will be removed because the decoder on line 70 will be unable to decode it and the tx will not be included in the final list of tx

Reply:
That makes sense, we should be error logging in those scenarios anyway